src/client.rs

```diff
@@ -1360,35 +1413,38 @@ impl Client {
     /// Pushes a single chunk of a blob to a registry,
     /// as part of a chunked blob upload.
     ///
-    /// Returns the URL location for the next chunk
+    /// Returns the URL location for the next chunk, alongside the start of the next range to upload
     async fn push_chunk(
         &self,
         location: &str,
         image: &Reference,
         blob_data: &[u8],
-        start_byte: usize,
+        range_start: usize,
     ) -> Result<(String, usize)> {
         if blob_data.is_empty() {
             return Err(OciDistributionError::PushNoDataError);
         };
-        let end_byte = if (start_byte + self.push_chunk_size) < blob_data.len() {
-            start_byte + self.push_chunk_size - 1
-        } else {
-            blob_data.len() - 1
-        };
-        let body = blob_data[start_byte..end_byte + 1].to_vec();
+
+        let range_size = self.push_chunk_size.min(blob_data.len());
+        let end_range_inclusive = range_start + range_size - 1;
+
+        let body = blob_data[..range_size].to_vec();
```

Comment:
I don't know how I missed this the first time, but this shouldn't use `to_vec()`. This means you'd end up cloning each chunk of data as it is pushed. The http request body should be able to just take the slice made here

Reply:
We have to provide something that is compatible with [`RequestBuilder.body`](https://docs.rs/reqwest/0.12.22/reqwest/struct.RequestBuilder.html#method.body).

The [`Body`](https://docs.rs/reqwest/0.12.22/reqwest/struct.Body.html) struct has the `From` trait implemented for: `&'static [u8]`, `&'static str`, `Bytes`, `File`, `String` and `Vec<u8>`.

I think in our case, the only viable solution would be to move to [`Bytes`](https://docs.rs/bytes/1.10.1/bytes/struct.Bytes.html)

Reply:
I believe the last commit should fix that up. I described the changes in the commit message